azerothcore/azerothcore-wotlk
# fix(core/QuestHandler) question marks not shown

## Changes Proposed:

This PR proposes changes to:

## Issues Addressed:

- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/10553
- Closes https://github.com/chromiecraft/chromiecraft/issues/2915
- Closes https://github.com/chromiecraft/chromiecraft/issues/7392

## Tests Performed:

This PR has been:

## How to Test the Changes:


1. To reproduce the error, just take a mission, whose npc contains a gossip. After taking the mission and interacting with the NPC, to see the menu, you move far enough away from it until it disappears from your sight. When you approach him again, you will see that the sign above his head disappears, thus avoiding showing you the status of the mission (whether complete or incomplete). The only way to see the sign again is to exit the game and re-enter it. At that moment, the sign appears again.
2. After applying the pull request, perform the same action, and you will see that no matter how far you go, the sign will always remain above the npc's head, thus showing not only who to give the mission to, but also that also, its status (complete or incomplete)

![WoWScrnShot_093024_130254](https://github.com/user-attachments/assets/855d0de2-c18e-4167-b3ce-92626a813175)

![WoWScrnShot_093024_130130](https://github.com/user-attachments/assets/ba5dcf6d-3055-413f-8d0e-42a771807950)

![WoWScrnShot_093024_130132](https://github.com/user-attachments/assets/f33e646c-0b41-41a6-9a4a-ef398334db92)

![WoWScrnShot_093024_130144](https://github.com/user-attachments/assets/ab2ae0ee-57af-4061-be7c-7e4598fa0368)

![WoWScrnShot_093024_131006](https://github.com/user-attachments/assets/3865d8d6-1898-49cc-90e1-bbf9e6d1675b)

![WoWScrnShot_093024_131019](https://github.com/user-attachments/assets/55f37dd6-2e25-41ed-8a0b-3821e964e6e7)


## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
